src/renderers/shared/stack/reconciler/ReactRef.js

```diff
@@ -43,7 +50,10 @@ ReactRef.attachRefs = function(instance, element) {
   }
 };
 
-ReactRef.shouldUpdateRefs = function(prevElement, nextElement) {
+ReactRef.shouldUpdateRefs = function(
+  prevElement: ReactElement | null | false,
+  nextElement: ReactElement | null | false,
```

Comment:
I think these can be `number | string` too, can they not?
In case where you have `<div>{1}{'2'}</div>` so they each get an instance.